backend/src/services/instance/instance.entity.ts

```diff
@@ -0,0 +1,121 @@
+import { getRandomId } from '~/helpers/helpers';
+
+class Instance {
+  public id: string;
+  public name: string;
+  public createdAt: string;
+  public keyPairId: string;
```

Comment:
Ideally It's better to have here `keyPair: KeyPairEntity`. The reason for it that instance **owns** keypair. It means that if we want to edit/delete/... keypair **of** instance we should do something like that: 
```
const instance = await repository.getById(...);
const keyPair = await service.create(...);
instance.addKeyPair(keyPair: KeyPair)
await repository.save(instance);
```
or:
```
const instance = await repository.getById(...);
const user = currentUser;
instance.deleteKeyPair(user: User) {
   // for example, in deleteKeyPair method we can enable deleting only for admins
   if (user.role != Role.Admin) throw Exception(...)
}
await repository.save(instance);
```
or:
```
const instance = await repository.getById(...);
const isConnectable = instance.isConnectable() {
   // here we can check if key pair exists and is also active. It may be overkill for our app, but in realworld it can be definitely a case
   return this.keyPair?.isActive;
}
```
**NOTE:** All these examples I have provided is just for your information. You **don't need** to edit any existing code. Really, having entity in entity can be harder to implement and we are now trying to deliver features as fast as possible. I just hope that they exmplained some more reasonings behind Entities.